### ISSUE TITLE
Should we add https://github.com/prettier/prettier ?

### ISSUE THREAD

**Issue by masters3d** [exercism/typescript#55]

**Comment by dsudia** [comment]
@masters3d when in the process would it run? do you mean just for our tests and examples, or as something that modifies learners' code on save? I think I'd be fine with it for our code (just so any exercises that get added keep a consistent format) but not sure that I'd be comfortable messing with someone's style. Maybe could include it as a script they could run if they want to...

**Comment by masters3d** [comment]
The project should have a style that is independent of the person contributing. This is similar to the way the linter works.

**Comment by dsudia** [comment]
@masters3d sure I completely agree for the track itself. But are you suggesting it run on someone's solution they are going to submit as well? I think I phrased that poorly above.

**Comment by dsudia** [comment]
@masters3d trip report on prettier. We've been using it at work for several months now and have run into an issue where it will reformat files on someone's branch that they haven't worked on (we run it on every file before commit) so we end up with PRs with 80 files changed, even if the person has only actually changed two files. If anyone has run into this and/or has a fix I'd be interested not just for this project but also to take back to my team!

**Comment by masters3d** [comment]
The same happens with other formation tools if they are run on the whole project. This can also happened if a new style rule is introduced in a newer version. I am not sure if its worth it. Probably just TSLint would be fine for now but I haven't tried it yet.

**Comment by dsudia** [comment]
yeah. The issue we had with prettier is that we've been locked on the same version with the same config file for months and this happens. I tried a branch where I branched off master, ran it, merged all the changes back into master, then made a new branch and ran it, and it made 90 more changes (mostly tiny things like "move this thing that prettier just broke into two lines on the last branch back into one line"). It felt very unstable. I upgraded it to the newest version and ran into the same issues, albeit less often.

Basically, it felt super useful for getting a project that has strayed way off into a useful state in one go (as opposed to doing hundreds or thousands of little formatting changes manually) but doesn't seem super stable as a long term solution yet.

**Comment by masters3d** [comment]
I was just thinking it could be its own command like `yarn pretty` in the same way that we have `yarn lint`. It only matters for people who are going to submit a new exercise to the repo. 

I do think that it may be too new to implement because there seems to be new changes coming out frequently. 

Maybe we just wait it out for a little longer.

**Comment by jrr** [comment]
Kind of related, could/should we include a `.vscode/settings.json` with `"editor.formatOnSave": true`? This was one of the first things I did in my own repo when I started on the typescript track.

Can a given language track cause files to be placed at the level above individual exercises? (so we could have one .vscode directory as a sibling to the exercises, instead of inside each exercise)
